Study Guide agent

BrandonwLii opened the pull request:
Closes #214 

Added a study guide agent to examples that loads .txt and .pdf documents located in agentic/examples/study/docs. The agent uses the weaviate rag tool.

The study guide agent has 3 main "modes":

- Test Prep: creates a short-answer test and answers
- Study Buddy: answers questions about the content in the RAG
- Multiple Choice Quiz: generates a multiple choice quiz and an answer key

The agent provides citations whenever RAG content is used, and is prompted to use general knowledge as little as possible.

Here is a segment from the openstax Calculus textbook (free) that I used for testing:
[calc1.pdf](https://github.com/user-attachments/files/21340249/calc1.pdf)

scottpersinger commented on examples/study/study_guide.py:
```diff
@@ -0,0 +1,84 @@
+
+from agentic.common import Agent, AgentRunner
+from agentic.models import GPT_4O_MINI
+# This is the "hello world" agent example. A simple agent with a tool for getting weather reports.
```
Suggest you update the comment with a fuller description of this agent and how it works. Note your PR comment will be lost to history, but this comment in code will live forever.

The comment should explain how to setup and run the agent. Also , does the RAG indexing happen every time or only once?

BrandonwLii commented:
Just updated the documentation for the agent, and added it into the examples list. Also changed the multiple choice agent to generate answers with the questions (hopefully reducing odds of hallucination)
